### PR TITLE
Removed duplicate code, implemented autoload for seeders and run migrations

### PR DESCRIPTION
**Pull request by RodolfoSilva**
I know it's not much, but let's leave so temporarily.

### CODE REVIEW

File: application/tests/_ci_phpunit_test/functions.php [hunk 1]
```diff
@@ -52,27 +52,8 @@ function reset_instance()
...
-	load_class('Input', 'core');
-	load_class('Lang', 'core');
-	
-	$loader = new CITEST_Loader();
-	load_class_instance('Loader', $loader);
-	
-	$controller = new CI_Controller();
-	return $controller;
+	reset_instance();
+	return get_instance();
```
Comment (kenjis) [inline application/tests/_ci_phpunit_test/functions.php]: This code is the same as before? It seems `new CI_Controller()` is missing.

Comment (RodolfoSilva) [inline application/tests/_ci_phpunit_test/functions.php]: Yes, but returning now current instance of CI.

**Comment by kenjis** [comment]
I removed duplicate code: https://github.com/kenjis/ci-phpunit-test/commit/095f8762bd5872dd66e1dc0b7ffa1beb3c71cbe9

In my understanding, if we don't `new CI_Controller()`, CI instance has still old state.

**Comment by kenjis** [comment]
Autoloading seeder and migration has your own logic (eg, from CIUnit specifications) and they will be tightly coupled stronger.

So I don't merge this PR.

I added CIPHPUnitTestCase class and now TestCase class is empty.
You can add your own logic to TestCase class.

**Comment by RodolfoSilva** [comment]
Yes, you're right.

**Comment by RodolfoSilva** [comment]
You will do the merge of `Seeder.php`. 
I'll create a new PR with modifications of `Seeder.php`.
I can do this?

**Comment by kenjis** [comment]
### Migration 

Why do you have to run migration in Seeder?
The logic can move to TestCase class.

### _test database

> only load if used database ends on '_test'

I think I don't need the limitation any more. We can separate test database `config/testing/database.php`

**Comment by RodolfoSilva** [comment]
Yes, you're right again.
We have add TestCase.php for custom user class in README.md

**Pull request closed without merging**